### DISCUSSION TITLE
Is it possible to configure a Zigbee dimmer module to work with a rocker switch?

### DISCUSSION THREAD

**Discussion by ndbroadbent** [Koenkk/zigbee2mqtt#12684] [Devices]
I have some Moes Zigbee dimmer modules, and I would prefer to use my existing rocker switches to just turn the lights on and off. I don't mind if I can't control the brightness using my wall switch. I still like how the LED lights fade in and out nicely. And I can still control the brightness using Home Assistant automations, Alexa, Siri, etc.

I have a mix of the following dimmer modules:

* [Lonsonho TS110E_1gang](https://www.zigbee2mqtt.io/devices/TS110E_1gang.html)
* [Lonsonho TS110E_2gang](https://www.zigbee2mqtt.io/devices/TS110E_2gang.html#lonsonho-ts110e_2gang)
* [Moes MS-105B](https://www.zigbee2mqtt.io/devices/MS-105B.html#moes-ms-105b)
* [Moes MS-104Z](https://www.zigbee2mqtt.io/devices/MS-104Z.html#moes-ms-104z)
* [Moes MS-105Z](https://www.zigbee2mqtt.io/devices/MS-105Z.html#moes-ms-105z)

They're designed to work with a push button, where a short press toggles them on/off and a long press changes the brightness. I'm trying to see if I can disable this behavior and just get them to turn on and off.

Similar discussions:

* https://github.com/Koenkk/zigbee2mqtt/discussions/12549
* https://github.com/Koenkk/zigbee2mqtt/discussions/11668
* https://github.com/Koenkk/zigbee2mqtt/discussions/9933
* https://github.com/Koenkk/zigbee2mqtt/discussions/5568
* https://github.com/Koenkk/zigbee2mqtt/discussions/11732

Thanks for your help!

**Comment by Matriciel** [comment]
https://www.zigbee2mqtt.io/devices/CCT5010-0001.html
This module is dimmer, with rocker, without neutral wire and ... router !!

**Comment by gregoryduckworth** [comment]
Did you get anyway with this? Realised I've gone down the same rabbit hole. I was thinking it might be possible to update the zigbee cluster to remove the functionality, but I know nothing about how to configure it.

**Reply by ndbroadbent** [reply]
Unfortunately I wasn’t able to figure this out, so I just replaced all of our rocker switches with momentary switches. I’d still be interested to find a solution though, I think the rocker switches are a bit easier to use and less confusing for guests.

**Reply by ndbroadbent** [reply]
This sounds very promising: https://github.com/Koenkk/zigbee2mqtt/issues/12952

<!DOCTYPE html><ol dir="auto" style="box-sizing: border-box; padding-left: 2em; margin-top: 0px; margin-bottom: 16px; caret-color: rgb(36, 41, 47); color: rgb(36, 41, 47); font-family: -apple-system, BlinkMacSystemFont, &quot;Segoe UI&quot;, Helvetica, Arial, sans-serif, &quot;Apple Color Emoji&quot;, &quot;Segoe UI Emoji&quot;; font-size: 14px; font-style: normal; font-variant-caps: normal; font-weight: 400; letter-spacing: normal; orphans: auto; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: auto; word-spacing: 0px; -webkit-tap-highlight-color: rgba(26, 26, 26, 0.3); -webkit-text-size-adjust: 100%; -webkit-text-stroke-width: 0px; text-decoration: none;"><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;"><p dir="auto" style="box-sizing: border-box; margin-top: 16px; margin-bottom: 16px;">It also has ability to configure the switch type on per-endpoint base. The<span class="Apple-converted-space"> </span><code class="notranslate" style="box-sizing: border-box; font-family: ui-monospace, SFMono-Regular, &quot;SF Mono&quot;, Menlo, Consolas, &quot;Liberation Mono&quot;, monospace; font-size: 11.899999618530273px; padding: 0.2em 0.4em; margin: 0px; background-color: var(--color-neutral-muted); border-radius: 6px;">Read Attributes</code><span class="Apple-converted-space"> </span>response is consistent.</p>

Field | Value
-- | --
Cluster | genLevelCtrl (0x0008)
Command | Write Attributes (0x02)
Attribute | 0xfc02
Data Type | 8-Bit Unsigned Integer (0x20)
Possible values | 0 - momentary ("kick-back switch"), 1 - toggle ("Seesaw Toggle Switch"), 2 - state ("Seesaw Sync Switch")

</li><li style="box-sizing: border-box; margin-top: 0.25em; margin-left: 0px;"></p></li></ol><br class="Apple-interchange-newline">

It sounds like the firmware probably does have this option, so there’s probably a way to set it from Z2M. I’ve seen some developer tools and low-level settings where you can send raw commands, but I’m not too sure how to do that

**Reply by gregoryduckworth** [reply]
I did some more digging on the Moes 2 gang dimmer and it looks like it isn't supported on the device itself.

**Reply by ndbroadbent** [reply]
Ah that’s a shame. Can I ask how did you find this information? I haven’t been able to find any manuals or documentation that describe which Zigbee features and commands are supported

**Reply by gregoryduckworth** [reply]
I actually used `zha` to find out the functionality of the device itself. It reported back as ```0x16d9:1] Discovered endpoint information: SizePrefixedSimpleDescriptor(endpoint=1, profile=260, device_type=81, device_version=1, input_clusters=[0, 4, 5, 61184], output_clusters=[25, 10])``` where in this case the `output_clusters` is important and refers to `Time` and `OTA`, so there is no external switch type like on the Lonsonho model.

**Reply by ndbroadbent** [reply]
Ah ok, thanks! In that case I suppose the only other option would be to write a custom firmware and figure out how to flash it. I don't think I would be able to do figure that out though! So might just stick with my momentary switches for now. Thanks for your help!

**Comment by ndbroadbent** [comment]
Oh wow, I just realized that a “switch type” option was added a few weeks after I posted this issue: https://github.com/Koenkk/zigbee2mqtt/issues/12952

I can confirm this is now available and working for my Lonsonho TS110E_2gang dimmer modules:

![075DB710-02E0-4B72-AA9C-7B7C6D64CE5C](https://user-images.githubusercontent.com/139536/191998283-abf31058-b25c-4734-a4f1-86511ae6d68d.jpeg)

I set it to “toggle”, and now my normal rocker switches are able to turn the lights on and off. That’s awesome!

Now that I’ve seen how they did it in this PR, I should be able to modify and install a fork to try adding this option for my Moes dimmer modules. I might try that this weekend

**Reply by Alejandroid17** [reply]
What is the status of this? I have the same problem :(. I am using Moes MS-105BZ and I don't know if I should return them.

**Reply by Trickytree1984** [reply]
Same for me Moes MS-105BZ using Zigbee2MQTT and i don't have the option for "switch_type" in settings to set it to a toggle switch

**Reply by Trickytree1984** [reply]
Actually the Moes MS-105BZ pairs as a Tuya TS0601 in  Zigbee2MQTT

**Reply by FPSUsername** [reply]
Hi, I bought the same [TS110E_2gang](https://www.zigbee2mqtt.io/devices/TS110E_2gang_2.html) dimmer module. It works fine, except I don't have the `switch_type` available.

I can set the power on behavior, turn on/off the lights, set min, max brightness and the current brightness.

In the exposes tab, the power on behavior seems to only target L1 instead of both (or have two separate entries). Seems like a bug.

```json
{
    "brightness_l1": 255,
    "brightness_l2": 255,
    "light_type": "led",
    "linkquality": 152,
    "max_brightness": 255,
    "max_brightness_l1": 255,
    "max_brightness_l2": 255,
    "min_brightness": 80,
    "min_brightness_l1": 65,
    "min_brightness_l2": 80,
    "power_on_behavior": "off",
    "power_on_behavior_l1": "off",
    "power_on_behavior_l2": "previous",
    "state_l1": "OFF",
    "state_l2": "OFF"
}
```

Seems like an issue:
https://github.com/Koenkk/zigbee2mqtt/issues/24788

**Comment by tvdbroeck** [comment]
I added a Lonsonho/Girier `TS110E` device to my network. I have the 1 gang version (manufacturer `_TZ3210_zxbtub8r`). I noticed that the convertor for the 1 gang version has a lot less features than the convertor for the 2 gang version. Even though these features are also supported by the 1 gang version. Specifically I was missing the tuya_switch_power_outage_memory and the TS110E_switch_type in the 1 gang version. The switch type was very important, because without it, the manual switch behaviour was unusable.
I made the following convertor, and this seems to work fine in my installation:

```
const definition = {
    fingerprint: [{modelID: 'TS110E', manufacturerName: '_TZ3210_zxbtub8r'}, {modelID: 'TS110E', manufacturerName: '_TZ3210_ngqk6jia'}],
    model: 'TS110E_1gang',
    vendor: 'Lonsonho',
    description: 'Zigbee smart dimmer module 1 gang with neutral',
    fromZigbee: extend.light_onoff_brightness({disablePowerOnBehavior: true}).fromZigbee.concat([
        fz.tuya_switch_power_outage_memory, fzLocal.TS110E_switch_type]),
    toZigbee: extend.light_onoff_brightness({disablePowerOnBehavior: true}).toZigbee.concat([
        tz.tuya_switch_power_outage_memory, tzLocal.TS110E_switch_type]),
    exposes: [
        e.light_brightness(),
        exposes.presets.switch_type_2(),
        exposes.enum('power_outage_memory', ea.ALL, ['on', 'off', 'restore']).withDescription('Recover state after power outage')
    ]
};
```
Is it possible to change this convertor on github, so other users can benefit as well?

Two additional remarks:


- the brand of my '1 gang dimmer' is Girier (https://nl.aliexpress.com/item/1005003693644741.html), but in the supported devices page (https://www.zigbee2mqtt.io/devices/TS110E_1gang.html), only the Lonsonho brand is mentioned. Could you add the Girier brand, so other users can more easily find it back?

- the 1 gang dimmer also seems to suffer from this bug (losing connectivity in 1 direction after a while): https://github.com/Koenkk/zigbee2mqtt/issues/12774
Creating a dummy group with only this device, seems to solve the problem. Not sure what causes this, but could be useful to mention on the documentation page.
Thanks for all the hard work!

**Comment by zycarlc** [comment]
I am late to this. But I wanted to leave some comments so people won't waste their time like me.

For the Moes device (ZM-105B-M I used), I tried to implement an external_converter with expose of `tuya.exposes.switchType()` which is used by other dimmers. Then, the switch type is exposed as expected. 

However, changing the switch type doesn't make any effect of the physical switch.

So, the best way is to avoid these devices and go with others that have been proven working with regular switch..

**Reply by toine512** [reply]
Too late for me.
I can confirm it doesn't work!

**Comment by toine512** [comment]
To sum up:
Sunricher and Schneider Electric ones explicitely require a momentary switch.
Tuya TS110E (Lonsonho, Girier) supports rocker switches (3 modes selectable) but may or may not be unuseable due to networking issues.
TS0601 is a Tuya radio with UART interface. That means the manufacturer of the device implements whatever interface through Tuya datapoints they want. I have confirmation from MOES support that ZM-105 models only support a momentary switch.

What's left?

**Reply by FPSUsername** [reply]
I have the ts110e and it works fine, only issue with the 2 gang dimmer is that it sometimes reports incorrect state on L1 (it randomly shows as turned off after a while).

Honestly I wish Sonoff could make one. I have the zbminir2 switch and it works nicely (and has OTA updates)

**Reply by toine512** [reply]
Looking at #12774, the existence of an issue seems completely random. So, good for you if you're lucky. Thanks for sharing. I bought the Girier kind, we'll see.

I wish Sonoff made one as well. I have the ZBMINI: connects immediately when powered up, never misses a command. I run the pilot wire of my 2 electric heaters with them.

**Reply by Wraithers** [reply]
I was looking at getting the [2 gang Moes ZB-105B-M-MS](https://www.zigbee2mqtt.io/devices/ZM-105B-M.html) (Tuya TS0601?) from their [website](https://moeshouse.com/products/zigbee-smart-dimmer-switch-module-timer?variant=47134032757051) and in my research came across this thread.

From what I can understand, the issue is that using it with a rocker switch only works to change the brightness of a connected light, but not toggle it on or off, correct?

If so, would it be possible in Home Assistant/ openHAB (if one is using them) to create a proxy dimmer and exclusively use that to control the dimming? Then when an action is detected at the rocker switch we set that to instead set the brightness to 0 if non-zero and vice versa?

I'm not familiar with the MQTT messages this model broadcasts, so I don't know if we can differentiate between changes via a proxy dimmer and its own state messages when activated via a rocker switch, therefore this may not be feasible. It would be great to find out from anyone on this thread if it might be 🤞

Edit: Fixed MoesHouse link to 2 gang variant

**Reply by FPSUsername** [reply]
The moes 105b does indeed not support it. It's easier to get the tuya dimmer from AliExpress. Unfortunately the 2 gang dimmer has a bug that sends an off state of L1. When you toggle it on, it sometimes resets the brightness.

Edit:
Take the Avatto ZDMS16 instead. It's reliable compared to the TS110E dimmers with the zbdongle-e running version 8.0.2 ember.

**Reply by Wraithers** [reply]
> If so, would it be possible in Home Assistant/ openHAB (if one is using them) to create a proxy dimmer and exclusively use that to control the dimming? Then when an action is detected at the rocker switch we set that to instead set the brightness to 0 if non-zero and vice versa?
> 
> I'm not familiar with the MQTT messages this model broadcasts, so I don't know if we can differentiate between changes via a proxy dimmer and its own state messages when activated via a rocker switch, therefore this may not be feasible. It would be great to find out from anyone on this thread if it might be 🤞

What about the proxy suggestion I made 👆 Could that work..?
@toine512 Do you still have one to test something like this?

**Comment by Pierre-Lavoine** [comment]
Hi there,

I’m going crazy with this module:
https://moeshouse.com/products/zigbee-smart-dimmer-switch-module-timer?variant=47134032691515

Setup:

Zigbee2MQTT (Docker) + Home Assistant (Docker)

Device is MOES ZM-105-M-MS (1 gang)

Detected as: TS0601 / _TZE200_dcnsggvz

Z2M model: TS0601_dimmer_5

Issue #1 — rocker switch behavior is erratic
I’m using a physical rocker (toggle) switch on the input.
Even when I set the device to “toggle/state” (switch_type), the behavior is unpredictable:

sometimes the light turns OFF as expected,

sometimes it starts dimming up/down instead,

sometimes brightness drops to 0% and then jumps back to 1% for no apparent reason.

It feels like the device interprets the rocker action as short/long presses randomly, or ignores the switch_type setting.

Issue #2 — excessive LQI traffic (“spamming” the Zigbee network)
This device publishes LQI extremely frequently (at least ~4 messages per second), which unnecessarily increases Zigbee traffic.

Questions:

Is rocker (toggle) switching actually supported by this device/firmware, or is it effectively “momentary only” regardless of switch_type?

Is there a way to reduce/disable the frequent LQI reporting (or any known workaround)?

Thanks!
